### PR TITLE
CDS-57043-propagate services between stage templates

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md [hunk 1]
```diff
@@ -49,6 +49,115 @@ You can also use Harness input sets and overlays to select from different collec
...
+
+:::
+
+When defining a pipeline, you can pass the service, its variables, artifact and manifest inputs through multiple stages.
+
+Following are the supported service propagation combinations: 
+
+- A non-template CD stage with subsequent non-templated CD stages
+- A templated CD stage with subsequent templated CD stages (stages referring the same template) 
+- A templated CD stage with subsequent templated CD stages (stages referring a different template)
```
Comment (yogesh-chauhan) [inline docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md:66]: We support all combinations here. The stage ( which propagates service from a previous stage ) can be a templated or not templated one.
Same goes for the stage from which the service is propagated. That means the type of the 2 stages are independent of each other

**Review by yogesh-chauhan — APPROVED** [review]

File: docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md [hunk 2]
```diff
@@ -49,6 +49,106 @@ You can also use Harness input sets and overlays to select from different collec
...
+
+### Limitations
+
+You cannot propagate services between different deployment types. For example, you cannot propagate a Kubernetes service between a Kubernetes deployment stage and a Native Helm deployment stage.
+
+### Requirements
+
+Make sure that the service is configured as a runtime input in the stage template or stage.
+
+### Visual summary
```
Comment (StrawberryPad) [inline docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md:70]: "Visual summary" is such an unnatural expression for a video. How about including "video" in the heading?

File: docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md [hunk 3]
```diff
@@ -49,6 +49,106 @@ You can also use Harness input sets and overlays to select from different collec
...
+
+1. Open or create a pipeline with two stage templates. For more information, go to [create a stage template](/docs/continuous-delivery/x-platform-cd-features/templates/create-a-remote-stage-template).
+2. Select the **Service** tab if it is not already selected.
+   
+   The propagation option appears.
+
+   ![](static/template-stage-propagation1.png)
+3. Select **Propagate from** and then select the stage with the service you want to use.
+   
+   You can see the stage name and service name.
```
Comment (StrawberryPad) [inline docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md:86]: I can only see the stage name in the image... maybe circle the service name in red?

File: docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md [hunk 4]
```diff
@@ -49,6 +49,106 @@ You can also use Harness input sets and overlays to select from different collec
...
+
+   ![](static/template-stage-propagation2.png)
+
+   You can also propagate a service from one template stage to another if it is a runtime input. 
+
+Here is a sample service propagation YAML
```
Comment (StrawberryPad) [inline docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md:92]:
```suggestion
Here is a sample service propagation YAML:
```

File: docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md [hunk 5]
```diff
@@ -49,6 +49,106 @@ You can also use Harness input sets and overlays to select from different collec
...
+            spec:
+              service:
+                useFromStage:
+                  stage: Deploy_Dev
+
+```
+
 ## Multiple service deployment
 
 :::info
```
Comment (StrawberryPad) [inline docs/continuous-delivery/x-platform-cd-features/services/propagate-and-override-cd-services.md:154]: This heading has very little info, so a note doesn't seem needed here, just a couple of short paragraphs.

**Review by StrawberryPad — CHANGES_REQUESTED** [review]
Looks good! Just a couple of minor issues.

**Review by thisrohangupta — APPROVED** [review]

**Pull request merged**